platforms/android/library-compose/src/main/java/io/element/android/wysiwyg/compose/RichTextEditor.kt

```diff
@@ -131,8 +139,11 @@ private fun RealEditor(
 
         view
     }, update = { view ->
-        view.setStyleConfig(style.toStyleConfig(view.context))
-        view.applyStyle(style)
+        if (style != previousStyle) {
+            view.setStyleConfig(style.toStyleConfig(view.context))
```

Comment:
I wrapped the `mentionDisplayHandler` parameter in a lambda to make it stable and prevent recompositions, but there's still some weird issue with `style` and `textStyleTypeface`, where they won't cause recompositions when the code is build from a clean state, but if *anything* in the code changes and we re-build, the `update` block will recompose with *every* change in the composer.

Reply:
@jonnyandrew could you double check if you can also reproduce the issue above? If not, it might just be an issue on my side and we can assume this is fixed.

Reply:
Very odd - I can reproduce this unfortunately.

Reply:
Then I'm not sure what we should do with this. It looks like some kind of Compose Compiler bug? Should we re-use the workaround of `previousStyle` to prevent unnecessary re-renderings of the HTML?

Reply:
If I understood correctly, the issue shouldn't affect the production library because it's a clean build, so a workaround shouldn't be necessary. But DevX on the other hand, becomes a nightmare and risky if we cannot check the behaviour is correct without doing a clean build.

I can see two paths forward:
- isolate the the change that causes this then at least we'd be able to exclude it manually during development by commenting out code. Or,
- add a UI test that can ensure the `update` block is not called more often than it should (something we should probably do at some point regardless).

Reply:
Sorry, I forgot to mention this: https://github.com/matrix-org/matrix-rich-text-editor/pull/854/commits/817e5bbf03d40da873794fd98c7b206ee68453dd actually solves this issue by using:

```kotlin
update = remember(capturedArgs...) { 
	{
		// Actual update function
	}
}
```

I added the `previousStyle != style` workaround for `EditorStyledText` to prevent the `HtmlConverter` from being re-created for each `text` change though.

Reply:
Now that we have a bit more time to work with, is there any way we can eliminate this workaround?

Reply:
It depends on whether we want to avoid creating a new `HtmlConverter` every time anything changes in this component or not. The `remember` makes the update more predictable, but it will still be triggered when any of its params change.

The current `AndroidHtmlConverter` is quite light-weight since it only creates a lambda, so we could allow this, or even create the converter on demand whenever it's supposed to be used, but this might not be the case for future or alternative implementations.

Reply:
Ok, I'll remove the `previousStyle` check, but the `remember` will have to stay for now, I don't really see what's causing the lambda to be re-created...

Reply:
Actually, `previousStyle` has already been removed 😅 . I completely forgot about that.